scrapy_autounit/middleware.py

```diff
@@ -70,15 +72,15 @@ def process_spider_input(self, response, spider):
         filter_args = {'crawler', 'settings', 'start_urls'}
         if isinstance(spider, CrawlSpider):
             filter_args |= {'rules', '_rules'}
-        response.meta['_autounit'] = {
+        response.meta['_autounit'] = pickle.dumps({
```

Comment:
This actually is a bit out of scope but not that much either.
Turns out that if the callback modified `request.meta` for instance, we were recording the modified version in the end (in `process_spider_output`).
This way we record the request just as it arrived to the callback.